packages/echo/src/client.ts

```diff
@@ -72,15 +72,47 @@ export class Echo {
   public static NOVU_SIGNATURE_HEADER = HttpHeaderKeysEnum.SIGNATURE;
 
   constructor(config?: ClientConfig) {
-    this.apiKey = config?.apiKey;
-    this.backendUrl = config?.backendUrl ?? DEFAULT_NOVU_API_BASE_URL;
-    this.devModeBypassAuthentication = config?.devModeBypassAuthentication || false;
+    const builtConfiguration = this.buildConfiguration(config);
+    this.apiKey = builtConfiguration.apiKey;
+    this.backendUrl = builtConfiguration.backendUrl;
+    this.devModeBypassAuthentication = builtConfiguration.devModeBypassAuthentication;
 
     const ajv = new Ajv({ useDefaults: true });
     addFormats(ajv);
     this.ajv = ajv;
   }
 
+  private buildConfiguration(config?: ClientConfig) {
+    const builtConfiguration: { apiKey?: string; backendUrl: string; devModeBypassAuthentication: boolean } = {
+      apiKey: undefined,
+      backendUrl: DEFAULT_NOVU_API_BASE_URL,
+      devModeBypassAuthentication: process.env.NODE_ENV === 'development',
+    };
+
+    Object.entries(builtConfiguration).reduce((_acc, [prop]) => {
+      const envVar = `NOVU_${toConstantCase(prop)}`;
+      if (config && config[prop]) {
+        builtConfiguration[prop] = config[prop];
+      } else if (process.env[envVar]) {
+        builtConfiguration[prop] = process.env[envVar];
+      }
+
+      return '';
+    }, '');
+
+    return builtConfiguration;
+  }
+
+  public addWorkflows(workflows: Array<DiscoverWorkflowOutput>) {
+    for (const workflow of workflows) {
+      if (this.discoveredWorkflows.some((existing) => existing.workflowId === workflow.workflowId)) {
+        throw new WorkflowAlreadyExistsError(workflow.workflowId);
```

Comment:
Having said that, instead of an error class per error type it's better DX to have a NovuFrameworkError and add messages and reason. That way it's easier for consumers to consume a single type and for us it creates less code.

Reply:
you think that if the user wants to react on specific error states that matching message content is the way to go? in my experience, we do a catch all to log and then re throw, while particular errors that interest us are better targeted by matching types as opposed to message content

Reply:
In TS, every new type is an extra import. So the catch-all code becomes cumbersome. My suggestion wasn't just a message. It combines message, reason, and code in a `NovuError` class. The code is what helps you dictate logic. Let me know if this helps.

Reply:
yep, that makes more sense.

Reply:
Ok so I took the time to look at the situation more. Basically all errors inherit up the tree the `NovuError` base class, and this class includes a http status code as well as a code that describes what happened from the framework perspective.

GIven how try/catch works, with this inheritance approach the user can aim for exactly the specific error type they want with `instanceof`. If they want to work with multiple error type imports and handle each case that way, they can. On the other hand, they can simply aim for `NovuError` and see its code to decide what to do next.

Does this make sense @SokratisVidros ?

Reply:
Exactly right @denis-kralj-novu , the `NovuError` implements a generic interface, the extended Errors are just sugar for explicitly declaring errors throughout the framework. This approach grants them flexibility to handle different types of errors with `instanceof`, whilst also granting flexibility to use the `NovuError.code` to discriminate on the error if their preference is for error object access instead.